### PR TITLE
DAOS-14352 object: fetch ec boundary

### PR DESCRIPTION
**Pull request by wangdi1**
Fetch EC boundary epoch before object migration actively, since during reintegration, EC boundary may not be synced to the reintegrating target before object migration.

Required-githooks: true
Features: rebuild ec

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by wangdi1** [comment]
Since it is run by features: rebuild, ec, so failed by DAOS-14320 and DAOS-14242. ping

File: src/container/container_iv.c [hunk 1]
```diff
@@ -1071,6 +1075,33 @@ cont_iv_ec_agg_eph_refresh(void *ns, uuid_t cont_uuid, daos_epoch_t eph)
...
+ds_cont_fetch_ec_agg_boundary(void *ns, uuid_t cont_uuid)
+{
+
+	struct cont_iv_entry	iv_entry = { 0 };
+	int			rc;
+
+	/* Only happens on xstream 0 */
+	D_ASSERT(dss_get_module_info()->dmi_xs_id == 0);
+	uuid_copy(iv_entry.cont_uuid, cont_uuid);
+	rc = crt_group_rank(NULL, &iv_entry.iv_agg_eph.rank);
```
Comment (liuxuezhao) [inline src/container/container_iv.c:1088]: seems the iv_agg_eph.rank not really needed for IV_CONT_AGG_EPOCH_BOUNDRY, so may need not get rank here.

**Review by liuxuezhao — APPROVED** [review]

**Review by liw — APPROVED** [review]

**Comment by daltonbohning** [comment]
@wangdi1 This PR introduced a codespell error: https://github.com/daos-stack/daos/actions/runs/6540435665

**Comment by wangdi1** [comment]
> @wangdi1 This PR introduced a codespell error: https://github.com/daos-stack/daos/actions/runs/6540435665

Ah, sure, will fix this in the following patch.  Thanks! @daltonbohning

**Pull request merged**